main.go

```diff
@@ -95,37 +96,36 @@ func main() {
 	}
 }
 
-func run(ctx context.Context, r *resources.GroupsClient, ttl time.Duration, dryRun bool) error {
+func run(ctx context.Context, r *armresources.ResourceGroupsClient, ttl time.Duration, dryRun bool) error {
 	log.Println("Scanning for stale resource groups")
-	listComplete, err := r.ListComplete(ctx, "", nil)
-	if err != nil {
-		return fmt.Errorf("Error when listing all resource groups: %v", err)
-	}
-	for list := listComplete; list.NotDone(); err = list.Next() {
+
+	pager := r.NewListPager(nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
 		if err != nil {
-			return fmt.Errorf("Error when iterating resource groups: %v", err)
+			log.Fatalf("failed to advance page: %v", err)
```

Comment:
```suggestion
			return fmt.Errorf("failed to advance page: %w", err)
```

Return an error here so `fatal` is only in the caller?